addons/stock_picking_batch/models/stock_picking.py

```diff
@@ -1,14 +1,27 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import _, api, fields, models
+from odoo import _, api, Command, fields, models
+from odoo.osv import expression
+from odoo.exceptions import ValidationError
 
 
 class StockPickingType(models.Model):
     _inherit = "stock.picking.type"
 
     count_picking_batch = fields.Integer(compute='_compute_picking_count')
     count_picking_wave = fields.Integer(compute='_compute_picking_count')
+    auto_batch = fields.Boolean('Automatic Batches', default=False,
+                                help="Automatically put pickings into batches as they are confirmed when possible.")
+    batch_group_by_partner = fields.Boolean('Contact', default=False, help="Automatically group batches by contacts.")
+    batch_group_by_destination = fields.Boolean('Destination Country', default=False, help="Automatically group batches by destination country.")
+    batch_group_by_src_loc = fields.Boolean('Source Location', default=False,
+                                            help="Automatically group batches by their source location.")
+    batch_group_by_dest_loc = fields.Boolean('Destination Location', default=False,
+                                             help="Automatically group batches by their destination location.")
+    batch_max_lines = fields.Integer("Maximum lines per batch")
+    batch_max_pickings = fields.Integer("Maximum transfers per batch")
```

Comment:
Since the default is to show "0" in the form view, maybe a help description saying something like "A transfer will not be added to batches that will exceed this number of lines if the transfer is added to it. Leave this value as '0' if no line limit." would be useful. Of course something similar should also be adapted for the `batch_max_pickings` and `batch_max_weight`

Reply:
Sounds good, done.